wpilibcExamples/src/main/cpp/examples/MecanumDrivePoseEstimator/cpp/Drivetrain.cpp

```diff
@@ -0,0 +1,55 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "Drivetrain.h"
+
+#include <frc2/Timer.h>
+
+#include "ExampleGlobalMeasurementSensor.h"
+
+frc::MecanumDriveWheelSpeeds Drivetrain::GetCurrentState() const {
+  return {units::meters_per_second_t(m_frontLeftEncoder.GetRate()),
+          units::meters_per_second_t(m_frontRightEncoder.GetRate()),
+          units::meters_per_second_t(m_backLeftEncoder.GetRate()),
+          units::meters_per_second_t(m_backRightEncoder.GetRate())};
+}
+
+void Drivetrain::SetSpeeds(const frc::MecanumDriveWheelSpeeds& wheelSpeeds) {
+  std::function<void(units::meters_per_second_t, const frc::Encoder&, frc2::PIDController&, frc::PWMVictorSPX&)> calcAndSetSpeeds = [&m_feedforward = m_feedforward](units::meters_per_second_t speed, const auto &encoder, auto &controller, auto &motor) {
```

Comment:
`[&m_feedforward = m_feedforward]` seems redundant. Does `[&m_feedforward]` not work?

Reply:
It does not afaik. See: https://stackoverflow.com/a/7896206 and https://stackoverflow.com/a/42029220